### ISSUE TITLE
Syntax error in async for comprehension depending on newline

### ISSUE THREAD

**Issue by Kazy** [davidhalter/parso#139]
Hello,

Using the latest master of `parso`, with Python 3.7.6, in an async for comprehension, it reports a `SyntaxError` depending on whether or not there is a newline before `]` (in the case of a list comprehension).

This is accepted:
```python
async def go():
    return [1 
        async for _ in range(5)]
```

But this isn't even though CPython is fine with it:
```python
async def go():
    return [1 
        async for _ in range(5)
    ]
```

I first experienced it with Jedi and I've reproduced it in `parso` like this:
```python
import parso

script_success = """
async def go():
    return [1 async for _ in range(5)]
"""

script_success_2 = """
async def go():
    return [1 
        async for _ in range(5)]
"""

script_fail = """
async def go():
    return [1 
        async for _ in range(5)
    ]
"""

grammar = parso.load_grammar()
for module in map(grammar.parse, [script_success, script_success_2, script_fail]):
    errors = grammar.iter_errors(module)
    print(errors)
    for err in errors:
        print(err.message)

# Output:
# []
# []
# [<Issue: 901>, <Issue: 901>]
# SyntaxError: invalid syntax
# SyntaxError: invalid syntax
```

Thanks !

**Comment by davidhalter** [comment]
@Kazy Thanks for the report! How are you using parso? Just wondering, because not a lot of people are using this feature as of now.

**Comment by Kazy** [comment]
@davidhalter With Jedi. More precisely, I'm using [Jedi Language Server](https://github.com/pappasam/jedi-language-server) with [diagnostic-nvim](https://github.com/nvim-lua/diagnostic-nvim) and [completion-nvim](https://github.com/nvim-lua/completion-nvim), and I experienced a bug where there were two `SyntaxError` on the last two line of a similar async for comprehension. I reproduced it with Jedi and since I already knew that `parso` was handling the parsing for Jedi, I tried to reproduce the bug in `parso` directly.
